### PR TITLE
Improve code coverage for System.Diagnostics.TextWriterTraceListener

### LINKED ISSUE #924: Improve code coverage for System.Diagnostics.TextWriterTraceListener
According to the current code coverage report, this library is only at 73%.  Our general goal is closer to 90%, though this will depend on the library.  We should investigate to see whether the current coverage level is appropriate, and if not, improve our test suites to provide better coverage.

### PR DESCRIPTION
**Pull request by Maxwe11**
Improve code coverage for `System.Diagnostics.TextWriterTraceListener` up to 89%

Closes #924

### CODE REVIEW

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

File: src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs [hunk 1]
```diff
@@ -1,88 +1,151 @@
...
+            builder.Append(delimiter);
+            builder.Append(eventType.ToString());
+            builder.Append(delimiter);
+            builder.Append(id.ToString(CultureInfo.InvariantCulture));
+            builder.Append(delimiter);
+            builder.Append(delimiter);
+            builder.Append(EscapedString(data.ToString()));
+            builder.Append(delimiter);
+
+            if (cache != null)
```
Comment (stephentoub) [inline src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs]: There's a lot of duplication between these two ExpectedTrace*Output functions.  Is it not possible to share most of the logic?  For example, the output from the cache section looks identical.

File: src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs [hunk 2]
```diff
@@ -1,88 +1,151 @@
...
+                    builder.Append(delimiter);
+            }
+
+            return builder.AppendLine().ToString();
+        }
+
+        private static string EscapedString(string str)
+        {
+            int index;
+ 	        int lastindex = 0;
```
Comment (stephentoub) [inline src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs]: Nit: some extra spaces on this line and one of the ones below

File: src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs (hunk 1 above)
Comment (Maxwe11) [inline src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs]: fixed

File: src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs (hunk 2 above)
Comment (Maxwe11) [inline src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs]: fixed

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@karelz, any feedback?

File: src/System.Diagnostics.TextWriterTraceListener/tests/CtorsStreamTests.cs [hunk 3]
```diff
@@ -1,143 +1,115 @@
...
+            Assert.Throws<ArgumentNullException>(() => new TextWriterTraceListener(stream: null));
 
-            //[] Stream is closed
-            _stream = new MemoryStream();
-            _stream.Dispose();
-            Assert.Throws<ArgumentException>(delegate () { new TextWriterTraceListener(_stream); });
+            var closedStream = new MemoryStream();
+            // We should call Close method here but for some reason we receive
+            // error CS1061: 'System.IO.MemoryStream' does not contain a definition for 'Close'.
+            // Calling Dispose should have same effect as calling Close.
```
Comment (Maxwe11) [inline src/System.Diagnostics.TextWriterTraceListener/tests/CtorsStreamTests.cs]: @karelz, amended so comments got deleted.
FYI you can leave persistent comments (stable for amending/rebasing)  through the "File Changes" tab on this page.

**Pull request merged**